h2o-genmodel/src/main/java/hex/genmodel/algos/tree/SharedTreeNode.java

```diff
@@ -13,7 +15,8 @@
  * Node in a tree.
  * A node (optionally) contains left and right edges to the left and right child nodes.
  */
-public class SharedTreeNode {
+public class SharedTreeNode implements INode<double[]>, INodeStat {
+  final int internalId; // internal tree id (that links the node back to the array of nodes of the tree) - don't confuse with user-facing nodeNumber!
```

Comment:
Is the array sequantially numberes ? If yes, this could be exposed via Tree API ?